lib/ansible/executor/task_executor.py

```diff
@@ -817,6 +810,33 @@ def _evaluate_failed_when_result(result):
         display.debug("attempt loop complete, returning result")
         return result
 
+    @staticmethod
+    def _retries_requested(_task):  # type: (Task) -> bool
+        return (_task.retries is not None) or bool(_task.until)
+
+    @staticmethod
+    def process_retry_parameters(retries_requested, _task):  # type: (bool, Task) -> Tuple[int, int]
+        '''
+        Extract the parameters used in retrying the task
+        '''
+        attempts = 1
+        if retries_requested:
+            if _task.retries is not None:
+                attempts += _task.retries
+            else:
+                attempts += 3
+
+        if attempts <= 0:  # guard against negative retries
+            attempts = 1
+
+        delay = _task.delay
+        if delay is None:
+            delay = Task.delay.default
```

Comment:
This does not seem right to me. The default should be automatically set by the field attribute, no?

Reply:
added in [this commit](https://github.com/lilatomic/ansible/commit/0e408e15159cfce2ed88543e72a4e79202acaf1b) to support `delay: null`, in response to [this comment](https://github.com/ansible/ansible/pull/76101#discussion_r733923981) after discovering that not all invalid delays would be corrected to valid delays.